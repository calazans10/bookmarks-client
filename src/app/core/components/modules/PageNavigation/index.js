import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './index.scss';

const PageNavigation = ({ pathname, title }) => (
  <div className="page-navigation">
    <ul className="page-navigation__list">
      <li className="page-navigation__item">
        <Link to={pathname}>{title}</Link>
      </li>
    </ul>
  </div>
);

PageNavigation.propTypes = {
  pathname: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default PageNavigation;
