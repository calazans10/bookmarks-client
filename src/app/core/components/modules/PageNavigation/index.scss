@import 'styles/utilities';

.page-navigation {
  margin-bottom: rem(25);

  @media(min-width: 600px) {
    margin-top: rem(60);
  }
}
