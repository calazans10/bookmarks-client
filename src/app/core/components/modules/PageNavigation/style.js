import styled from 'styled-components';
import { rem } from 'polished';

export const Container = styled.div`
  margin-bottom: ${rem('25px')};

  @media (min-width: 600px) {
    margin-top: ${rem('60px')};
  }
`;
