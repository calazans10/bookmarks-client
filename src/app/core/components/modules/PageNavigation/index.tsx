import React from 'react';
import { Link } from 'react-router-dom';
import { Container } from './style';

type PageNavigationProps = {
  pathname: string;
  title: string;
};

const PageNavigation = ({ pathname, title }: PageNavigationProps) => (
  <Container>
    <ul>
      <li>
        <Link to={pathname}>{title}</Link>
      </li>
    </ul>
  </Container>
);

export default PageNavigation;
