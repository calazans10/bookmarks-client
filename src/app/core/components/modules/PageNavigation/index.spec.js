import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PageNavigation from './index';

describe('PageNavigation', () => {
  it('renders without crashing', () => {
    const props = {
      pathname: '/home',
      title: 'Home',
    };
    const wrapper = shallow(<PageNavigation {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
