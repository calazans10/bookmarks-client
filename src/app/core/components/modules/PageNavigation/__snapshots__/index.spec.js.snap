// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNavigation renders without crashing 1`] = `
<div
  className="page-navigation"
>
  <ul
    className="page-navigation__list"
  >
    <li
      className="page-navigation__item"
    >
      <Link
        to="/home"
      >
        Home
      </Link>
    </li>
  </ul>
</div>
`;
